src/transformers/tokenization_utils.py

```diff
@@ -1096,42 +1140,53 @@ def batch_encode_plus(
         # Compute longest sequence size
         max_seq_len = max(map(len, batch_outputs["input_ids"]))
 
-        if return_attention_masks:
-            # Allow the model to not give any special attention to padded input
-            batch_outputs["attention_mask"] = [[0] * len(v) for v in batch_outputs["input_ids"]]
-
         if return_tensors is not None:
 
             # Do the tensor conversion in batch
             for key, value in batch_outputs.items():
-
-                padded_value = value
-                # verify that the tokenizer has a pad_token_id
-                if key != "input_len" and self._pad_token is not None:
-                    # Padding handle
-                    padded_value = [
-                        v + [self.pad_token_id if key == "input_ids" else 1] * (max_seq_len - len(v))
-                        for v in padded_value
-                    ]
-
                 if return_tensors == "tf" and is_tf_available():
-                    batch_outputs[key] = tf.constant(padded_value)
+                    try:
+                        batch_outputs[key] = tf.constant(value)
+                    except ValueError:
+                        if None in [item for sequence in value for item in sequence]:
+                            raise ValueError(
+                                "No padding token is set for this model, therefore no batch can be made"
```

Comment:
NO_PAD_TOKEN_MSG = "" as a module level constant will make this smaller.

Reply:
You're right, that would be better indeed!